--- /usr/share/X11/xkb/symbols/sk	2013-02-03 01:52:10.000000000 +0100
+++ /usr/share/X11/xkb/symbols/sk2	2013-02-19 09:44:34.000000000 +0100
@@ -101,3 +101,45 @@
 
     key <BKSL>	{ [ backslash,        bar,        slash,     NoSymbol ]	};
 };
+
+partial alphanumeric_keys
+xkb_symbols "dvorak-ucw" {
+
+    // US Dvorak layout extend with UCW - AltGr produces accented characters
+
+    include "us(dvorak)"
+
+    name[Group1]= "Slovak (US Dvorak with CZ UCW support)";
+
+    // (AB03) 
+    // A-alphanumeric
+    // row counted from bottom of the keyboard (B = second)
+    // number of keyboard from left (03)
+
+    key <AB01> { [ semicolon, colon, adiaeresis, Adiaeresis ] };
+    key <AB02> { [ q, Q, ocircumflex, Ocircumflex ] };
+    key <AB03> { [ j, J, EuroSign, NoSymbol ] };
+    key <AB04> { [ k, K, uacute, Uacute ] };
+    key <AB10> { [ z, Z, zcaron, Zcaron ] };
+
+    key <AC01> { [ a, A, aacute, Aacute ] };
+    key <AC02> { [ o, O, oacute, Oacute ] };
+
+
+    key <AC03> { [ e, E, eacute, Ecaron ] };
+    key <AC04> { [ u, U, uacute, Uacute ] };
+    key <AC05> { [ i, I, iacute, Iacute ] };
+    key <AC06> { [ d, D, dcaron, Dcaron ] };
+    key <AC08> { [ t, T, tcaron, Tcaron ] };
+    key <AC09> { [ n, N, ncaron, Ncaron ] };
+    key <AC10> { [ s, S, scaron, Scaron ] };
+
+    key <AD05> { [ y, Y, yacute, Yacute ] };
+    key <AD08> { [ c, C, ccaron, Ccaron ] };
+    key <AD09> { [ r, R, racute, Racute ] };
+    key <AD10> { [ l, L, lacute, Lacute ] };
+    key <AD11> { [ slash, question, lcaron, Lcaron ] };
+
+    include "level3(ralt_switch)"
+};
+
